hedera-node/src/main/java/com/hedera/services/state/expiry/ExpiringCreations.java

```diff
@@ -20,33 +20,52 @@
  * ‍
  */
 
+import com.google.protobuf.ByteString;
+import com.hedera.services.context.ServicesContext;
 import com.hedera.services.context.properties.GlobalDynamicProperties;
+import com.hedera.services.legacy.core.jproto.TxnReceipt;
 import com.hedera.services.records.RecordCache;
 import com.hedera.services.state.EntityCreator;
 import com.hedera.services.state.merkle.MerkleAccount;
 import com.hedera.services.state.merkle.MerkleEntityId;
+import com.hedera.services.state.submerkle.CurrencyAdjustments;
+import com.hedera.services.state.submerkle.EntityId;
 import com.hedera.services.state.submerkle.ExpirableTxnRecord;
+import com.hedera.services.state.submerkle.RichInstant;
+import com.hedera.services.state.submerkle.TxnId;
+import com.hedera.services.utils.TxnAccessor;
 import com.hederahashgraph.api.proto.java.AccountID;
-import com.hederahashgraph.api.proto.java.TransactionRecord;
+import com.hederahashgraph.api.proto.java.Timestamp;
+import com.hederahashgraph.api.proto.java.TokenTransferList;
+import com.hederahashgraph.api.proto.java.TransactionReceipt;
+import com.hederahashgraph.api.proto.java.TransferList;
 import com.swirlds.fcmap.FCMap;
 
+import java.time.Instant;
+import java.util.ArrayList;
+import java.util.List;
 import java.util.function.Supplier;
 
+import static com.hedera.services.state.submerkle.EntityId.fromGrpcScheduleId;
+import static com.hedera.services.utils.MiscUtils.asTimestamp;
+import static com.hederahashgraph.api.proto.java.ResponseCodeEnum.FAIL_INVALID;
+
 public class ExpiringCreations implements EntityCreator {
 	private RecordCache recordCache;
 
 	private final ExpiryManager expiries;
 	private final GlobalDynamicProperties dynamicProperties;
 	private final Supplier<FCMap<MerkleEntityId, MerkleAccount>> accounts;
+	private final ServicesContext ctx;
 
 	public ExpiringCreations(
 			ExpiryManager expiries,
 			GlobalDynamicProperties dynamicProperties,
-			Supplier<FCMap<MerkleEntityId, MerkleAccount>> accounts
-	) {
-		this.accounts = accounts;
+			ServicesContext ctx) {
```

Comment:
Whenever I inject the `ServicesContext` into a component, I prefix its name with `Aware`...it marks the components that are self-aware of the fact they are running inside a `ServicesContext`; and know how to get more of their collaborators _from_ that context.

Ideally none of our components will be `Aware`, and instead get ALL their dependencies through constructor injection. 

This will make it almost trivial for us to migrate to a REAL IoC container like [Dagger 2](https://dagger.dev/dev-guide/). :innocent:

Reply:
Understood. That sounds good. 💯 
In this case I removed the constructor injection, as `ServicesContext` is used only in one method. Instead passed it as a parameter to the method using it. Thanks Michael!